libs/sdk-core/src/errors.rs

```diff
@@ -0,0 +1,74 @@
+use reqwest::StatusCode;
+use std::time::Duration;
+
+pub type SdkResult<T, E = ExternalSdkError> = anyhow::Result<T, E>;
+
+// TODO Rename to SdkError once existing SDKError struct is replaced
+/// Public error kinds returned by the SDK
+#[derive(Debug)]
+pub enum ExternalSdkError {
+    LnInvoiceError { err: String },
```

Comment:
For generic errors I would try and define them without additional err data.
This way they will be specific and the user will know exactly what to do. Could be it will cause us to implement more specific errors.

Reply:
Done, I removed any internal error data for now. Instead, I tried to reflect the specific meaning of the error in the name.